practicals/2-od.md

```diff
@@ -0,0 +1,321 @@
+
+# 1 Review Homework
+
+You should now be familiar with the basics of R and the `tidyverse`. If
+you have not completed these tasks go back and do them first:
+
+- Read Chapters 2, 3, and 4 of [Reproducible road safety research with
+  R](https://itsleeds.github.io/rrsrr/basics.html)
+- Read Chapters 3 and 5 of [R for Data
+  Science](https://r4ds.had.co.nz/data-visualisation.html)
+
+# 2 Getting started with GIS in R
+
+Note that this practical takes sections from Chapters 2 - 8 of
+[Geocomputation with R](https://r.geocompx.org). You should expand your
+knowledge by reading these chapters in full.
+
+## Pre-requisites
+
+You need to have a number of packages installed and loaded. Install the
+packages by typing in the following commands into RStudio (you do not
+need to add the comments after the `#` symbol)
+
+If you need to install any of these packages use:
+
+``` r
+install.packages("sf")  # Install a package from CRAN
+remotes::install_github("Nowosad/spDataLarge")   # install from GitHub using the remotes package
+```
+
+``` r
+library(sf)          # vector data package 
+library(tidyverse)   # tidyverse packages
+```
+
+- It relies on **spData**, which loads datasets used in the code
+  examples of this chapter:
+
+``` r
+library(spData)  # spatial data package 
+```
+
+1.  Check your packages are up-to-date with `update.packages()`
+2.  Create an RStudio project with an appropriate name for this module
+    (e.g. `TDSmodule`)
+3.  Create appropriate folders for code, data and anything else
+    (e.g. images)
+4.  Create a script called `learning-OD.R`, e.g. with the following
+    command:
+
+``` r
+dir.create("code") # 
+file.edit("code/learning-OD.R")
+```
+
+## 2.1 Basic sf operations
+
+We will start with a simple map of the world. Load the `world` object
+from the `spData` package. Notice the use of `::` to say that you want
+the `world` object from the `spData` package.
+
+``` r
+world = spData::world
+```
+
+Use some basic R functions to explore the `world` object.
+e.g. `class(world)`, `dim(world)`, `head(world)`, `summary(world)`. Also
+view the `world` object by clicking on it in the Environment panel.
+
+`sf` objects can be plotted with `plot()`.
+
+``` r
+plot(world)
+```
+
+![](2-od_files/figure-gfm/unnamed-chunk-5-1.png)<!-- -->
+
+Note that this makes a map of each column in the data frame. Try some
+other plotting options
+
+``` r
+plot(world[3:6])
+```
+
+![](2-od_files/figure-gfm/unnamed-chunk-6-1.png)<!-- -->
+
+``` r
+plot(world["pop"])
+```
+
+![](2-od_files/figure-gfm/unnamed-chunk-6-2.png)<!-- -->
+
+## 2.2 Basic spatial operations
+
+Load the `nz` and `nz_height` datasets from the `spData` package.
+
+``` r
+nz = spData::nz
+nz_height = spData::nz_height
+```
+
+We can use `tidyverse` functions like `filter` and `select` on `sf`
+objects in the same way you did in Practical 1.
+
+``` r
+canterbury = nz %>% filter(Name == "Canterbury")
```

Comment:
Good content from Geocomputation with R.

Reply:
Between R4DS, RRSRR, and Geocomp with R they have a lot of good worked examples to go through. So I don't think we need to create more.